[7주차] Team 하니홈 신수진 & 원채영 미션 제출합니다.

chaeyoungwon opened the pull request:
## 배포 링크
[하니홈 투표 배포 링크 🔗](https://next-vote-21th.vercel.app/)

## 피그마 디자인 링크
[하니홈 투표 UI 🔗](https://www.figma.com/design/shLqxsAkJCA3MIoVzBH7OW/%ED%95%98%EB%8B%88%ED%99%88---%ED%88%AC%ED%91%9C-UI?node-id=1-3866)


## 과제를 하며

주어진 필수 구현 사항을 바탕으로 전체 UI를 디자인한 후, 이를 기준으로 파트를 나누었는데요. 유사한 UI끼리 묶어서 한 명은 홈, 로그인, 회원가입 페이지를, 다른 한 명은 멤버 페이지, 투표 페이지, 공통 모달을 맡기로 한 후, 사다리타기를 통해 역할을 정했습니다.

- **신수진: 프로젝트 세팅, 투표 페이지, 멤버 페이지, 공통 모달**
    
    > 채영이가 전반적인 디자인을 맡아서 하니홈 프로젝트 컨셉에 맞추어서 예쁘게 만들어주었습니다. 반응형으로 개발해야하다보니 디자인을 두 가지의 경우 모두 만들어주었는데, 너무 고생한거 같아서 박수 백 만 번 치고싶습니다. 👍🏻🔥👏🏻💓마지막 과제인 만큼 실력이 초반에 비해 조금 많이(?) 오른 것 같아 살짝은 뿌듯하지만, 아직 갈 길이 멀어보입니다…
    > 
    
    **[7주차 작업 내용]**
    
    - 백엔드와 프론트엔드 데이터를 각각 `BE`, `FE` 객체로 분리 저장하였고, 투표 페이지와 멤버 페이지에서 각 파트에 따라 URL 파라미터(`params.part`)를 사용하여 데이터를 동적으로 불러오는 구조로 구현하였습니다.
    - 공통 모달의 경우 `onClose`와 `onConfirm`두 개의 핸들러로 분리하여 역할 명확화하였습니다. 아니오(`onClose`)를 눌렀을 때 모달이 닫히고, 예(`onConfirm`)를 눌렀을 때 로직 실행 또는 페이지 이동 처리하였습니다. 재사용 가능한 컴포넌트로 설계하여 다른 투표 유형에서도 활용 가능하도록 구성하였습니다.
    - 반응형 레이아웃으로 구현하면서모바일 및 데스크탑에서 요소의 **정렬**, **크기**, **간격** 등이 자연스럽게 배치되도록 조정하였습니다.
    
    **[8주차 작업 내용]**
    - 파트장 및 데모데이 투표 결과 UI를 구현하고, 이에 투표 결과 조회 API를 연동하였습니다.
    - 동일한 스타일의 모달을 공통 컴포넌트로 분리한 뒤, 각 파일에서 import하여 사용하는 방식으로 리팩토링하였습니다.
    - 파트장 및 데모데이 투표 API를 연동하고, 같은 파트끼리만 투표가 가능하며, 같은 데모데이 팀에는 투표할 수 없도록 백엔드에서 제약을 걸어둔 로직에 맞춰, 해당 경우에는 백에서 전달되는 메시지를 기반으로 alert 처리를 추가하였습니다.
    - 사용자가 이미 투표한 파트장/팀 정보를 조회하는 API도 연동하여, 해당 항목에는 다른 스타일이 적용되도록 UI를 구현하였습니다.
    
- **원채영: UI 디자인, 공통 레이아웃, 홈, 로그인, 회원가입**
    
    > 하니홈 서비스의 디자인 시스템을 기반으로 작업을 진행했는데, 디자인 없는 개발자의 설움을 이제야 깨닫습니다.. 이번 과제를 통해 백엔드와의 협업을 미리 경험해볼 수 있어서 의미 있었고, 본격적인 연동 전 개발 흐름을 사전에 준비해볼 수 있는 좋은 기회였습니다.
    > 
    
    **[7주차 작업 내용]**
    
    - 브레이크포인트를 768px로 설정하여 반응형으로 개발하였습니다. 모바일 환경에서는 헤더 메뉴가 햄버거 버튼을 클릭했을 때 나타나는 내비게이션 구조로 변경하였고, 홈 화면에 사용된 배경 도형들의 위치도 모바일에 맞게 조정하였습니다.
    - 회원가입 폼에는 Zod 라이브러리를 사용하여 유효성 검사를 적용하였습니다. 입력값에 따라 에러 메시지를 출력하고 버튼을 비활성화하는 로직을 구현하여, 사용자 입력에 대한 에러 문구 출력을 구현하였습니다.
    - 로그인 및 회원가입 인증 처리에서는 Zustand를 활용하여 accessToken을 전역 상태로 관리하였습니다.
    - Axios 인스턴스를 설정하고, 요청 및 응답 인터셉터를 구성하여 accessToken이 만료되었을 경우 자동으로 갱신할 수 있는 구조를 구현하였습니다.
    
    **[8주차 작업 내용]**
    
    - 7주차에 받은 코드 리뷰를 바탕으로 전반적으로 코드를 리팩토링하고 깨지는 UI들을 수정하였습니다.
    - 홈 화면에 총 투표 수를 함수로 받아오며, 실시간으로 숫자가 올라가는 애니메이션을 추가하여 훅으로 구현하였습니다.
    - zod와 react-hook-form을 함께 사용하여 회원가입 시 거치는 유효성 검사 코드, 추론 코드 등을 개발하였습니다.
    이메일 및 아이디 중복 검사 API를 연결하여 회원가입 시 모든 입력과 API 연결에 성공하였을 경우 회원가입 버튼이 활성화되도록 조건을 추가했습니다.
    - 토큰 관리 방식을 구체화하고, 실제 로그인 및 회원가입, 로그아웃 API를 연결하였습니다.
    - 현재 백엔드에서는 토큰 만료 시 자동으로 재발급해주는 로직이 구현되어 있습니다.
        하지만 프론트엔드에서 Zustand를 사용해 토큰을 저장할 경우, 페이지를 새로고침하면 토큰이 초기화되는 문제가 발생합니다. 이를 해결하기 위해, 백엔드에 API 개발을 요청하여 페이지가 새로고침될 때 토큰 재발급 API를 호출하는 로직을 추가하였습니다.
        단, 사용자가 직접 로그아웃을 실행한 경우에는 재발급이 이루어지지 않도록 하기 위해, 로컬스토리지에 특정 값을 저장해 해당 상황을 구분하도록 처리했습니다..
        
    - middleware.ts에서 일반적으로 사용자 접근 제어를 처리하지만, 토큰을 메모리에 저장하고 있어
    쿠키에 저장된 리프레시 토큰의 존재 여부를 기반으로 로그인 여부를 판단하고자 했습니다. 하지만 도메인 불일치 등의 이유로 쿠키가 `undefined`로 전달되는 문제가 발생한 것으로 보였고,, 클라이언트 측에서 별도의 인증 훅을 작성하여 로그인 여부를 판단하고, 필요한 경우 라우팅을 제한하는 방식으로 처리하였습니다 ㅠㅅㅠ
    - 투표 결과를 득표순으로 정렬되어 있는 API가 따로 존재하여 기존에 연결해둔 API를 제거하고 해당 API로 대체하였습니다. 또한 중복되는 API 연결 로직을 하나로 통일하고, fetch로 연결되어 있던 API들은 axiosInstance를 사용하는 형태로 전반적으로 수정하였습니다.
    

## **Key Question**

### Zod 스키마가 무엇인지, 어떻게 활용할 수 있는지 알아봅시다.

`Zod`는 **런타임에서의 타입 검증**과 **스키마 기반 유효성 검사**를 지원하는 TypeScript 친화적 라이브러리

`import { z } from "zod";`

- 사용 이유: TypeScript는 **컴파일 타임**에만 타입을 검사할 수 있다. 하지만 실제 코드가 동작하는 시점은 런타임이며, 이때 잘못된 값이 들어오더라도 TS는 관여할 수 없다.
    
    ```tsx
    type User = { age: number }; // 컴파일 타임 제한만
    
    ```
    
    - `age`가 `string`이어도 실제 런타임에서는 에러 없이 통과할 수 있음
    - Zod를 쓰면 이런 상황에서 **런타임 검증이 가능**
        
        ```tsx
        const schema = z.object({ age: z.number().int().min(0) });
        
        schema.parse({ age: 21 });
        schema.parse({ age: "21" }); // 런타임 에러
        
        ```
        

---

- 장점
    - 런타임에서도 안전한 타입 보장
    - 스키마에서 TypeScript 타입 자동 추론
    - 숫자 범위, 문자열 형식 등 세밀한 유효성 검사 가능
    - 별도 DTO 타입을 정의하지 않아도 되는 일관성 유지

---

- 활용 예시
    
    ```tsx
    const signupSchema = z.object({
      id: z.string().min(6).max(20),
      email: z.string().email(),
      password: z.string().min(8).max(20),
    });
    
    try {
      signupSchema.parse(formData); // 성공 or 에러 throw
    } catch (e) {
      console.log(e.errors); // 유효성 에러 메시지 배열
    }
    
    ```
    

### 이번 프로젝트에서 토큰 관리를 어떻게 할 예정인지, 그리고 왜 그런 방법을 선택했는지에 대해 설명해 주세요.

> 이번 프로젝트에서는 Zustand 전역 상태 관리 라이브러리를 통해 access token을 관리하고, refresh token은 httpOnly 쿠키에 저장하여 자동 재발급 방식으로 인증 흐름을 구성할 예정입니다.
> 
> - access token은 상태로만 보관하여 컴포넌트 간 인증이 필요한 요청에 쉽게 활용할 수 있도록 합니다.
> - refresh token은 보안성을 고려해 httpOnly 쿠키에 저장하여 클라이언트 측에서 직접 접근하지 않도록 합니다.
> - access token이 만료되었을 때는, 서버에서 자동으로 refresh token을 활용해 access token을 응답 헤더에 담아 재발급해주는 구조입니다.

> 선택 이유
> 
> - 로컬스토리지에 access token을 저장하는 방식은 XSS 공격에 취약하다는 보안 우려가 있음
> - 쿠키에만 저장하는 방식은 상태 기반 인증 흐름 관리가 불편할 수 있음
> - 따라서 access token은 `Zustand`로 상태 관리하고, refresh token은 `httpOnly 쿠키`로 보호하며, 자동 재발급 구조를 통해 보안성과 사용 편의성 사이 균형을 맞추고자 함

lemoncurdyogurt committed:
environment: 초기 개발 환경 세팅

chaeyoungwon committed:
Merge pull request #1 from suyoungSwim/sujin

environment: 초기 개발 환경 세팅

chaeyoungwon committed:
chore: 디자인 시스템 및 svg 설정 추가

chaeyoungwon committed:
feat: 헤더 구조 추가

chaeyoungwon committed:
fix: svg 임포트 오류 해결

chaeyoungwon committed:
Merge branch 'master' into chaeyoung

chaeyoungwon committed:
feat: 헤더 및 홈 UI 1차 구현

chaeyoungwon committed:
feat: 홈페이지 UI 구현

chaeyoungwon committed:
feat: 로그인 페이지 UI 구현 및 구조 리팩토링

chaeyoungwon committed:
feat: 페이지에 들어와있을 경우 헤더 메뉴에 밑줄 추가

chaeyoungwon committed:
feat: 회원가입 페이지 1차 구현

chaeyoungwon committed:
feat: 커스텀 드롭다운 추가

chaeyoungwon committed:
refactor: 회원가입 컴포넌트 및 함수 분리, 비밀번호일 경우 인풋 필드에 눈 아이콘 추가

chaeyoungwon committed:
design: 인풋 필드 패딩 및 글자크기 조정

chaeyoungwon committed:
chore: pr 템플릿 추가

chaeyoungwon committed:
Merge branch 'master' into chaeyoung

chaeyoungwon committed:
chore: globals.css 수정 및 typography 설정 방식 변경

chaeyoungwon committed:
fix: 절대경로로 수정

lemoncurdyogurt committed:
merge: merge with chaeyoung branch

lemoncurdyogurt committed:
Merge branch 'chaeyoung' of https://github.com/suyoungSwim/next-vote-21th into sujin

lemoncurdyogurt committed:
fix: merge 오류 해결

lemoncurdyogurt committed:
feat: 로그인 모달 기능 추가

lemoncurdyogurt committed:
remove: 콘솔로그 삭제

lemoncurdyogurt committed:
feat: 로그인 모달 모바일 메뉴에 적용 및 반응형으로 사이즈 조정

lemoncurdyogurt committed:
feat: 회원가입 완료 모달 추가

lemoncurdyogurt committed:
style: 코드 prettier수행

lemoncurdyogurt committed:
chore: prettier script cache 기능 추가

lemoncurdyogurt committed:
feat: 회원가입 모달 화면에 띄우기

lemoncurdyogurt committed:
feat: vote modal 제작

lemoncurdyogurt committed:
feat: 투표 및 멤버 페이지 파일구조 생성

lemoncurdyogurt committed:
feat: 로그인 확인 모달 데스크탑 화면 메뉴바에도 추가

lemoncurdyogurt committed:
feat: 투표 및 멤버 페이지 파일구조 생성

lemoncurdyogurt committed:
feat: 멤버 데이터 추가

chaeyoungwon committed:
refactor: 코드리뷰 반영 코드 수정

chaeyoungwon committed:
✨ Feat: 홈, 로그인, 회원가입 UI 구현 (#2)

홈, 로그인, 회원가입 UI 구현

chaeyoungwon committed:
design: md 사이즈 내 글씨 크기 수정

chaeyoungwon committed:
feat: zod/zustand 설치 및 api 구조 연결

chaeyoungwon committed:
refactor: signup 페이지 내 구조 분리 및 훅 추가

chaeyoungwon committed:
fix: 로그아웃에 clearAuth 추가

chaeyoungwon committed:
✨ Feat: 로그인/회원가입 API 연결 구조 추가 및 zod 유효성 검사 적용 (#4)

로그인/회원가입 API 연결 구조 추가 및 zod 유효성 검사 적용

lemoncurdyogurt committed:
feat, design: 멤버페이지 추가, 반응형 디자인 적용

lemoncurdyogurt committed:
design: 스크롤바 hide

lemoncurdyogurt committed:
design: 멤버 페이지 디자인 수정

lemoncurdyogurt committed:
rename: 파트별로 페이지 동적라우팅

lemoncurdyogurt committed:
feat: 투표 처음 화면 추가

lemoncurdyogurt committed:
style: 코드 prettier적용

chaeyoungwon committed:
fix: 배포 에러 해결

chaeyoungwon committed:
fix: 배포 중 type 안맞는 이슈 해결

lemoncurdyogurt committed:
refactor: 데모데이 페이지 title추가

chaeyoungwon committed:
fix: 멤버 페이지 디자인 수정

chaeyoungwon committed:
fix: vote 첫 페이지 디자인 수정

chaeyoungwon committed:
fix: 절대경로로 변경 및 모달 배경색 tailwind 코드로 변경

chaeyoungwon committed:
fix: vote 첫 페이지 리팩토링 및 디자인 수정

lemoncurdyogurt committed:
design: 모달 버튼 색깔 수정

lemoncurdyogurt committed:
feat: 파트장 투표 페이지 추가

lemoncurdyogurt committed:
merge 충돌 해결

chaeyoungwon committed:
feat: 데모데이 투표하기 구현

chaeyoungwon committed:
Merge branch 'master' into sujin

chaeyoungwon committed:
fix: teamlist 구조 변경으로 인한 demoday 코드 수정

chaeyoungwon committed:
fix: params 오류 수정 및 폴더 구조 수정

chaeyoungwon committed:
fix: 회원가입 파일 충돌 오류 수정

lemoncurdyogurt committed:
fix: params 오류 수정

lemoncurdyogurt committed:
style: prettier 적용

lemoncurdyogurt committed:
merge: 충돌처리

lemoncurdyogurt committed:
design: 투표 페이지 디자인 설정

lemoncurdyogurt committed:
design: 투표 순위 텍스트 text align 설정

chaeyoungwon committed:
fix: 데모데이/ 파트장 투표하기 버튼 기능 동일하게 수정

chaeyoungwon committed:
fix: 투표 순위 보러 가기 문구 위치 수정

chaeyoungwon committed:
fix: svg fill 오류해결

chaeyoungwon committed:
✨ Feat: 공통 컴포넌트, 멤버 페이지, 투표선택 페이지 (#3)

공통 컴포넌트, 멤버 페이지, 투표선택 페이지 구현

chaeyoungwon committed:
fix: useAuthStore 내 변수 사용

chulheungkim commented on next-vote-21th/package.json:
```diff
@@ -0,0 +1,39 @@
+    "@svgr/webpack": "^8.1.0",
+    "axios": "^1.9.0",
+    "clsx": "^2.1.1",
+    "lucide-react": "^0.511.0",
```
찾아보니 Lucide 아이콘 라이브러리를 편리하게 사용할 수 있는 좋은 패키지인 것 같네요!
피그마에도 아이콘 Set들이 배포가 되어있어서 Material Design처럼 디자이너와 개발자가 동일한 아이콘을 쓰는데 큰 도움이 되는 플러그인인 것 같은데, 좋은 정보 알아갑니다!

chulheungkim commented on next-vote-21th/public/svgs/vote/violetShape.svg:
다양한 도형으로 다채로운 디자인을 만드셨네요,,,👏🏿

chulheungkim commented on next-vote-21th/src/app/layout.tsx:
```diff
@@ -0,0 +1,48 @@
+  icons: {
+    icon: "/svgs/favicon.svg",
+  },
+  openGraph: {
```
Next.js의 `Metadata`내 `openGraph` 속성을 사용해보지 않았는데, 하나 배워갑니다!

chulheungkim commented on next-vote-21th/src/app/login/page.tsx:
```diff
@@ -0,0 +1,87 @@
+              label="아이디"
+              placeholder="아이디를 입력해주세요."
+              value={form.id}
+              onChange={e => setForm({ ...form, id: e.target.value })}
```
이 `InputField` 컴포넌트에 넘기는 `onChange` 핸들러를 `key`(네이밍은 상관 x) 파라미터를 활용해 핸들러를 작성하고 `id`와 `password`를 매개변수로 넘겨주는 방식으로 리팩토링하면 좋을 것 같아요!

```tsx
type FormKey = keyof typeof form; // 'id' | 'password'

  const handleInputFieldChange = (key: FormKey, value: string) => {
    setForm({ ...form, [key]: value });
  };
```

chaeyoungwon replied:
넹 담번에 참고해서 작성해보겠습니다~

chulheungkim commented on next-vote-21th/src/app/members/page.tsx:
```diff
@@ -0,0 +1,59 @@
+
+const MembersPage = () => {
+  const [selectedPart, setSelectedPart] = useState("Front-End");
+  const parts = ["Front-End", "Back-End"];
```
`src/constants` 내에서 관리하면 더 좋을 것 같아요~

chulheungkim commented on next-vote-21th/src/app/members/page.tsx:
```diff
@@ -0,0 +1,59 @@
+import Profile from "@/public/svgs/members/profile.svg";
+
+const MembersPage = () => {
+  const [selectedPart, setSelectedPart] = useState("Front-End");
```
여기도 초기값을 `parts[0]`으로 하면 더 좋을 것 같습니당

chulheungkim commented on next-vote-21th/src/app/login/page.tsx:
```diff
@@ -0,0 +1,87 @@
+        </form>
+
+        <span className="text-cap1-med md:text-body2-med mt-[22px] text-center text-gray-800 underline">
+          <Link href="/signup">계정이 없다면 회원가입하러 가기 &gt;</Link>
```
right-arrow 아이콘을 `&gt;`로도 표현하는 방법도 상황에 따라 유용하게 쓰일 수 있겠네요!
하나 알아갑니다!

chulheungkim commented on next-vote-21th/src/app/page.tsx:
```diff
@@ -0,0 +1,68 @@
+
+const HomePage = () => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const isLoggedIn = false;
```
이 플래그는 상수 값으로 코드 내에서 값의 변화가 일어나지 않는데, 임시로 구현을 해놓은걸까요?!

chaeyoungwon replied:
아하 이 부분 stores 폴더에서 가져오도록 하려던 건데 코드 변경하다가 못 발견했네여.. 다음에 정리해서 반영해보겠습니다!

chulheungkim commented on next-vote-21th/src/components/home/BackgroundShape.tsx:
`transition-delay`로 간단하지만, 인상적인 애니메이션이 구현된 것 같네요!

chulheungkim commented on next-vote-21th/src/components/home/VoteCategoryList.tsx:
```diff
@@ -0,0 +1,22 @@
+  "# 프론트엔드 파트장 투표",
+  "# 백엔드 파트장 투표",
+  "# 데모데이 투표",
+];
```
보통 이러한 고정 값들을 하나의 파일에서 관리하는걸 선호하시는 편인가요?!

chaeyoungwon replied:
한 곳에서만 쓰이는 상수는 사용하는 컴포넌트 안에 정의해두면, 관련된 코드끼리 붙어 있어서 추적하기도 쉽고, 변경도 덜 번거롭더라구용

chulheungkim commented on next-vote-21th/src/schemas/signupSchema.ts:
확인을 해보니 zod도 v4가 출시됐네요! 아직 `.refine()` 메소드를 사용해서 Error Message를 제어할 수 있다는 것도 알아갑니다!

chulheungkim reviewed (commented):
전반적으로 구현이 잘 되어있는 것 같고, 과제하느라 고생 많으셨습니다!
개인적으로 `constants`와 `type`들의 분리를 통해 일관성 있는 재사용이 가능하게끔 리팩토링과 API 호출의 경우 Tanstack Query를 사용하면 error 메세지들과 상태 관리를 더 효율적으로 할 수 있을 것 같다는 생각이 드네요!

두분 다 고생 많으셨습니다!

2025314242 commented on next-vote-21th/src/app/layout.tsx:
```diff
@@ -0,0 +1,48 @@
+  display: "swap",
+  style: "normal",
+  weight: "45 920",
+  variable: "--font-suit",
```
localFont 적용하실 때 className 사용하신 거 같아서 variable property는 지우셔도 될 것 같습니다!

2025314242 commented on next-vote-21th/src/schemas/signupSchema.ts:
```diff
@@ -0,0 +1,22 @@
+      })
+      .refine(val => /(?=.*[0-9])(?=.*[a-zA-Z])(?=.*[\W_])/.test(val), {
+        message: "문자, 숫자, 특수문자를 모두 포함해주세요.",
+      }),
```
superRefine 메소드 사용하면 한 번에 분기처리할 수 있는 거로 알아서 한 번 찾아보시면 좋을 거 같아요! 별개로 스키마 타입 체킹이 깔끔해서 좋은 것 같습니다

2025314242 commented on next-vote-21th/src/types/signup/dto.ts:
```diff
@@ -0,0 +1,8 @@
+  email: string;
+  password: string;
+  confirmPassword: string;
+};
```
```export type SignupForm = z.infer<typeof signupSchema>;``` 이렇게 zod 자체에서 타입 추론할 수 있는 거로 아는데 한 번 적용해보시는 것도 좋을 거 같아요!

chaeyoungwon replied:
마지막에 zod를 추가하느라 전반적으로 사용법이 조금 미숙하다고 생각합니다.. 리팩토링 시 참고하겠습니다! 감사합니당

2025314242 commented on next-vote-21th/src/apis/auth.ts:
```diff
@@ -0,0 +1,65 @@
+  email: string;
+  position: "FRONTEND" | "BACKEND" | "DESIGN" | "PRODUCT_MANAGER";
+  team: "HANI_HOME" | "POP_UPCYCLE" | "DEAR_DREAM" | "PROMESA" | "INFLUY";
+};
```
저희는 아직 api 적용을 하지는 않았긴 한데, 이렇게 백으로 보내는 타입도 zod 스키마 적용할 수 있더라고요 extend나 omit, pick 같은 method 사용해서 적용하시면 좀 더 유연한 코드가 될 것 같습니다!

2025314242 commented on next-vote-21th/src/constants/menuItems.ts:
```diff
@@ -0,0 +1,35 @@
+
+type MenuItem =
+  | { label: string; href: string; onClick?: undefined }
+  | { label: string; href?: undefined; onClick: () => void };
```
never로 타입 명시하면 좀 더 strict해질 거 같습니다!

2025314242 commented on next-vote-21th/src/constants/menuItems.ts:
```diff
@@ -0,0 +1,35 @@
+    { label: "Vote", href: "/vote" },
+    { label: "Members", href: "/members" },
+  ];
+};
```
순서가 연관 있어서 Vote랑 Members를 중복 선언하신 건지 궁금합니다! 삼항 연산자 사용해도 괜찮을 거 같아요

chaeyoungwon replied:
그래도 괜찮을 거 같네욥 참고하겠습니다

2025314242 commented on next-vote-21th/src/hooks/useSignUpForm.ts:
```diff
@@ -0,0 +1,63 @@
+
+  const applyValidationErrors = (zodErrors: SignupErrors) => {
+    setErrors(zodErrors);
+  };
```
바로 setErrors 사용해도 될 거 같은데 혹시 따로 선언한 이유가 있을까요?

chaeyoungwon replied:
정신없던 개발의 흔적 같네요..........

2025314242 commented on next-vote-21th/src/components/signup/TeamSelector.tsx:
```diff
@@ -0,0 +1,53 @@
+
+interface TeamSelectorProps {
+  teams: readonly string[];
+  members: readonly string[];
```
좋은 옵션 배워갑니다 !

2025314242 reviewed (commented):
좋은 코드 감사합니다!

chaeyoungwon committed:
refactor: 코드리뷰 반영 코드 수정

chaeyoungwon committed:
♻️ refactor: 코드리뷰 기반 코드 수정 (#5)

🔥 작업 내용
- axiosInstance와 authInstance 분리하여 인증 여부에 따른 요청 구분
- zod + react-hook-form 적용 및 유효성 검사 로직 개선
- 불필요한 로컬 폰트의 variable 속성 제거
- zod.superRefine() 메서드 활용해 커스텀 검증 로직 작성
- 메뉴 아이템 조건부 렌더링을 삼항 연산자로 단순화하고, authItems로 그룹화
- memberData 구조 수정 및 하드코딩된 부분 정리하여 가독성 및 유지보수성 향상

chaeyoungwon committed:
fix: 엔드포인트 수정 및 중복확인 검사 주석처리

chaeyoungwon committed:
Merge branch 'master' into chaeyoung

chaeyoungwon committed:
refactor: 로그인 API tanstack query 이용 구조로 변경

chaeyoungwon committed:
refactor: id->username 변수명 변경

chaeyoungwon committed:
fix: api 구조 변경에 따른 isDuplicate -> exists로 변경

chaeyoungwon committed:
fix: 회원가입 검증 로직 수정

chaeyoungwon committed:
refactor: auth 관련 중복 로직 삭제

chaeyoungwon committed:
refactor: signup payload 타입형태로 import

chaeyoungwon committed:
design: 삭제된 gap 다시 추가

lemoncurdyogurt committed:
Merge branch 'CEOS-Developers:master' into master

lemoncurdyogurt committed:
style: 파트장 투표 화면 수정

lemoncurdyogurt committed:
refactor: 투표 완료후 투표 버튼 삭제

chaeyoungwon committed:
fix: exists 응답 오류 수정

chaeyoungwon committed:
🐛 Fix: 회원가입, 로그인 API 코드 구조 및 로직 수정 (#6)

chaeyoungwon committed:
fix: 배포 오류 수정

lemoncurdyogurt committed:
Merge branch 'master' of https://github.com/suyoungSwim/next-vote-21th into sujin

lemoncurdyogurt committed:
refactor: 모달 공통모달형식으로 코드 수정

lemoncurdyogurt committed:
style: 모달 텍스트 수정

lemoncurdyogurt committed:
refactor: functional component 수정

lemoncurdyogurt committed:
style: 파일 이름을 vote에서 elections로 수정

lemoncurdyogurt committed:
refactor: 파트별 타입 선언 분리

lemoncurdyogurt committed:
feat: 파트장 투표 결과 페이지 UI

lemoncurdyogurt committed:
feat: 투표순으로 정렬 기능 추가

lemoncurdyogurt committed:
feat: 데모데이 투표결과 UI

lemoncurdyogurt committed:
design: 데모데이 결과 디자인 수정

lemoncurdyogurt committed:
feat: 멤버ID저장

lemoncurdyogurt committed:
feat: candidate id 추가

lemoncurdyogurt committed:
feat: 이름으로 후보자ID찾는 유틸 추가

lemoncurdyogurt committed:
fix: id 타입에러 수정

lemoncurdyogurt committed:
fix: 투표 request body에서 memberId 제거로 불필요 코드 삭제

lemoncurdyogurt committed:
fix: 투표 post api 엔드포인트 수정

lemoncurdyogurt committed:
feat: 데모데이 및 파트장 투표 api 추가

lemoncurdyogurt committed:
feat: jwt decode 추가

lemoncurdyogurt committed:
refactor: 코드리뷰 기반 코드 수정

chaeyoungwon committed:
feat: middleware.ts 생성 및 리다이렉트 조건 추가

chaeyoungwon committed:
remove: 불필요한 주석 제거

chaeyoungwon committed:
✨ Feat: middleware.ts 파일 생성 및 리다이렉트 조건 추가 (#8)

chaeyoungwon committed:
feat: 새로고침 시 재발급 API 호출하여 토큰 저장 구현

chaeyoungwon committed:
✨ Feat: 새로고침 시 재발급 API 호출하여 토큰 저장 구현 (#9)

### 🔥 작업 내용
- ClientWrapper 생성 -> 새로고침 시 재발급 API 호출하여 토큰 저장 구현

chaeyoungwon committed:
feat: 로컬스토리지에 로그아웃 여부 저장

chaeyoungwon committed:
fix: 배포 오류 수정

lemoncurdyogurt committed:
design, test: 콘솔로그 제거 및 투표 하기 전 후보자 클릭 시 색상 수정

lemoncurdyogurt committed:
design: 스크롤바 숨김 및 backgroundshpae반영

chaeyoungwon committed:
Merge branch 'master' into sujin

chaeyoungwon committed:
fix: 충돌 코드 수정 및 axiosInstance로 수정

chaeyoungwon committed:
feat: 정렬된 후보 목록 조회 API로 로직 대체

chaeyoungwon committed:
rename: 파일명 수정

chaeyoungwon committed:
feat: 홈페이지 총 투표 수 함수 추가 및 애니메이션 훅 추가

chaeyoungwon committed:
✨Feat: 투표 결과 UI (#7)

chaeyoungwon committed:
fix: 배포 오류 수정

chaeyoungwon committed:
fix: any 타입 오류 수정

chaeyoungwon committed:
fix: alert 메시지 수정

chaeyoungwon committed:
design: 투표 결과 UI 디테일 수정

chaeyoungwon committed:
dd
